mgr/src/secrets.rs

```diff
@@ -68,3 +73,49 @@ impl Secrets {
         Ok(())
     }
 }
+
+pub fn generate_mnemonic_and_macaroons(secret_directory: &Path) -> Result<()> {
+    let mnemonic_path = secret_directory.join("mnemonic");
+    let mnemonic = if !mnemonic_path.exists() {
+        let mut rng = thread_rng();
+        let mnemonic = Mnemonic::generate_in_with(&mut rng, bip39::Language::English, 24)?;
+        fs::write(&mnemonic_path, mnemonic.to_string())
+            .with_context(|| format!("Cannot write to {}", mnemonic_path.display()))?;
+
+        println!("Generated a new mnemonic: {}", mnemonic_path.display());
+        mnemonic
+    } else if let Ok(words) = fs::read_to_string(mnemonic_path) {
+        Mnemonic::parse(words)?
+    } else {
+        panic!("mnemonic is incorrect")
+    };
+
+    let mut engine = sha256::HashEngine::default();
+    engine.input(&mnemonic.to_seed(""));
+    engine.input("macaroon/0".as_bytes());
+    let hash = sha256::Hash::from_engine(engine);
+    let seed = hash.into_inner();
+
+    let key = MacaroonKey::generate(&seed);
+    let mut admin_macaroon = Macaroon::create(None, &key, "admin".into())?;
+    admin_macaroon.add_first_party_caveat("roles = admin|readonly".into());
```

Comment:
The caveats on the macaroons could change, or new macaroons could be added. Its very configurable, they could change rarely or often depending on requirements. So they need to be generated on update not on install. Which means either the secret needs to be present for every update or we need a way to notify the user that the secret is required because a caveat changed or a new macaroon was added.

Reply:
I think the design is wrong. If a user wants to provide an existing seed on first install this is ok, it can be copied across to the server and read by KLD. But otherwise the seed and mnemonic should be generated by KLD and then copied to the workstation.

Reply:
Hmm, this depends on the architecture.  

@paulmattei Do you think the local machine running mgr should also have the `kld` on the local machine in order to generate the macaroons tokens, or we can use the same formula to generate macaroons as the current PR?

`mgr` does not bundle with `kld` now and I thought it will be an independent app to serve `kld`, `kneard`, and possibly more. (`kld-mgr` and `kneard-mgr` will possibly merge into one `mgr` app for all kuutamo products and move out).  If `mgr` need to bundle with `kld`, `kneard`, and more just to generate some token, `mgr` will be cumbersome.  

@JosephGoulden Because the `mgr` and `kld` will are separated, we can put secret as stand-alone share lib to generate macaroons for this, such that there will be one macaroon generated function and rescued by `kld` and `mgr`.

Reply:
Okay, let's resue MacaroonAuth in kld, and fix the way we build nix pkg for mgr.

Reply:
there won't be any new users so a diff isn't required, yet. Currently there isn't a way to create/delete users - until that exists it doesn't seem a concern.

Reply:
If `kld-mgr` is dependent on `kld`, the `kld-mgr` will easily break the MacOS support for not.  Another way is to split secret as a separate lib and we may do it when we need to create/delete users.

Reply:
To clarify thread, macaroons are still created by kld. This functionality is not changing. This PR adds that they are also created by the manager for local/workstation kld-cli use.